br/pkg/restore/log_client/compacted_file_strategy.go

```diff
@@ -37,19 +37,50 @@ func NewCompactedFileSplitStrategy(
 	}
 }
 
-func (cs *CompactedFileSplitStrategy) Accumulate(subCompaction *backuppb.LogFileSubcompaction) {
-	splitHelper, exist := cs.TableSplitter[subCompaction.Meta.TableId]
+type sstIdentity struct {
+	EffectiveID     int64
+	RewriteBoundary *utils.RewriteRules
+}
+
+func (cs *CompactedFileSplitStrategy) inspect(ssts SSTs) sstIdentity {
+	r, ok := ssts.(RewrittenSST)
+	if !ok {
+		return sstIdentity{
+			EffectiveID:     ssts.TableID(),
+			RewriteBoundary: nil,
+		}
+	}
+
+	rule := utils.GetRewriteRuleOfTable(ssts.TableID(), r.RewrittenTo(), 0, map[int64]int64{}, false)
+
+	return sstIdentity{
+		EffectiveID:     r.RewrittenTo(),
+		RewriteBoundary: rule,
+	}
+}
+
+func (cs *CompactedFileSplitStrategy) Accumulate(ssts SSTs) {
+	identity := cs.inspect(ssts)
+
+	splitHelper, exist := cs.TableSplitter[identity.EffectiveID]
 	if !exist {
 		splitHelper = split.NewSplitHelper()
-		cs.TableSplitter[subCompaction.Meta.TableId] = splitHelper
+		log.Info("Initialized splitter for table.",
+			zap.Int64("table-id", ssts.TableID()), zap.Int64("effective-id", identity.EffectiveID), zap.Stringer("rewrite-boundary", identity.RewriteBoundary))
+		cs.TableSplitter[identity.EffectiveID] = splitHelper
 	}
 
-	for _, f := range subCompaction.SstOutputs {
-		startKey := codec.EncodeBytes(nil, f.StartKey)
-		endKey := codec.EncodeBytes(nil, f.EndKey)
+	for _, f := range ssts.GetSSTs() {
+		startKey, endKey, err := utils.GetRewriteRawKeys(f, identity.RewriteBoundary)
+		if err != nil {
+			log.Panic("[unreachable] the rewrite rule doesn't match the SST file, this shouldn't happen...",
```

Comment:
A risk here if pitr command is exposed to sql statement later. it might panic the tidb instance. so far, it's ok.

Reply:
If I recall correctly TiDB captures panic for each connection, so hopefully this won't make the TiDB node crash